qmfkdldks/motion-menu-consumer
# Add review form page (consumer)

# What is the objective?

We want to build best UI and UX to improve current restaurant review system.

# User story

As a visitor, I want to leave a review for the shop so I can share my thoughts with others and express my appreciation.

# Acceptance criteria

- If user is not logged in, it shows login message
- If unauthenticated user press submit button, it redirect to first authenticate and then it submits automatically the review
- If the body has less than 3 characters, it shows an error message
- Review form text area is expandable (user can resize the height)
- When error happens while sending review create request, it shows error message
- On desktop index page shows review form.
- Redirect to the success page when everything is ok

# Technical Notes

- Mobile first design
- Desktop design combines two components. Review list and the review form (Comes later)

# What should be tested

- Submit button sends right request with params
- Unauthenticated user gets redirected and once it completes, it sends review POST request
- Text changes depending on authentication status
- Form is visible
- Redirect to the success page

# QA Notes

Explain how to test newly implemented features and what are steps required.

1. Visit the shop review page [https://url.com/](https://url.com)
2. Check form exists
3. Write invalid number of characters and check error and button is disabled
4. Submit as authenticated user, request should be ok
5. Submit as unauthenticated user, it should redrect to the sign up page

# Engineering Pitfalls

> If you have encountered some issues, you should leave notes here.
It will be used to prevent future errors. These notes can become our assets.
> 

# Measure Performance and Business

Questions to be answered

1. How many user want to leave reviews?
2. Why user leaves reviews?
    1. A survey is necessary (To be defined)

**Event and custom properties**

- review-form-page.review-form.submit
    - authenticated: boolean whehter user is authenticated or not
- review-form-page.visit (Is this necessary?)

**Monitor to be created**

User submits review / Number of visitors * 100 = Z%
